main.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"fmt"
+	"strconv"
+	"log"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type bin int
+
+func (b bin) String() string {
+	return fmt.Sprintf("%b", b)
+}
+
+func main() {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		panic(err.Error())
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	for {
+		pods, err := clientset.CoreV1().Pods("").List(metav1.ListOptions{})
+		if err != nil {
+			panic(err.Error())
+		}
+
+		restartThreshold := int32(100)
+
+		for i := 0; i < len(pods.Items); i++ {
+			for j := 0; j < len(pods.Items[i].Status.ContainerStatuses); j++ {
+				if pods.Items[i].Status.ContainerStatuses[j].State.Waiting != nil {
```

Comment:
It may be more readable to take the checks from lines 38-41 and extract them into a method. Just a suggestion.